speeduino/timers.ino

```diff
@@ -63,14 +85,20 @@ void oneMSInterval(void) //Most ARM chips can simply call a function
   bool isCrankLocked = configPage4.ignCranklock && (currentStatus.RPM < currentStatus.crankRPM); //Dwell limiter is disabled during cranking on setups using the locked cranking timing. WE HAVE to do the RPM check here as relying on the engine cranking bit can be potentially too slow in updating
   //Check first whether each spark output is currently on. Only check it's dwell time if it is
 
-  if(ignitionSchedule1.Status == RUNNING) { if( (ignitionSchedule1.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign1EndFunction(); ignitionSchedule1.Status = OFF; } }
-  if(ignitionSchedule2.Status == RUNNING) { if( (ignitionSchedule2.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign2EndFunction(); ignitionSchedule2.Status = OFF; } }
-  if(ignitionSchedule3.Status == RUNNING) { if( (ignitionSchedule3.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign3EndFunction(); ignitionSchedule3.Status = OFF; } }
-  if(ignitionSchedule4.Status == RUNNING) { if( (ignitionSchedule4.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign4EndFunction(); ignitionSchedule4.Status = OFF; } }
-  if(ignitionSchedule5.Status == RUNNING) { if( (ignitionSchedule5.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign5EndFunction(); ignitionSchedule5.Status = OFF; } }
-  if(ignitionSchedule6.Status == RUNNING) { if( (ignitionSchedule6.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign6EndFunction(); ignitionSchedule6.Status = OFF; } }
-  if(ignitionSchedule7.Status == RUNNING) { if( (ignitionSchedule7.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign7EndFunction(); ignitionSchedule7.Status = OFF; } }
-  if(ignitionSchedule8.Status == RUNNING) { if( (ignitionSchedule8.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ign8EndFunction(); ignitionSchedule8.Status = OFF; } }
+  if(ignitionSchedule1.Status == RUNNING) { if( (ignitionSchedule1.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ignitionSchedule1.pEndCallback(); ignitionSchedule1.Status = OFF; } }
```

Comment:
``  if(ignitionSchedule1.Status == RUNNING) { if( (ignitionSchedule1.startTime < targetOverdwellTime) && (configPage4.useDwellLim) && (isCrankLocked != true) ) { ignitionSchedule1.pEndCallback(); ignitionSchedule1.Status = OFF; } }
``
Please don't consider bit fields as bools, please change it to `(configPage4.useDwellLim == 1)` and as the C compiled code runs each check in sequence is better to use the limiter first like this:
``  if ((configPage4.useDwellLim == 1) && (isCrankLocked != true) && (ignitionSchedule1.Status == RUNNING) && (ignitionSchedule1.startTime < targetOverdwellTime)) { ignitionSchedule1.pEndCallback(); ignitionSchedule1.Status = OFF; }
``
This way if dwell limit is turned off the execution don't expend too much time, not critical, but is optimal, also avoiding nested `if` statements are better for clarity, I know this is from old code, but is better to fix it in a PR right? 😉

Reply:
> I know this is from old code, but is better to fix it in a PR right?

[Boy scout rule](https://matheus.ro/2017/12/11/clean-code-boy-scout-rule/) :wink:

If I make that change, I may as well hoist the the config check into an outer ` if ((configPage4.useDwellLim == 1) && (isCrankLocked != true) { /* Per channel checks here */ }`.

Reply:
Even better, the current way do the most important checks for last.

Reply:
And if doing that, I may as well encapsulate the per-channel check into an inline method:
```
static inline void applyOverDwellCheck(IgnitionSchedule &schedule, uint32_t targetOverdwellTime) {
  if ((schedule.Status == RUNNING) && (schedule.startTime < targetOverdwellTime)) { 
    schedule.pEndCallback(); schedule.Status = OFF; 
  }
}
```